### ISSUE TITLE
Ruby 2.3.x and older are not compatible with OpenSSL 1.1.x on Arch

### ISSUE THREAD

**Issue by BjornMelgaard** [rvm/rvm#4006]
1. Cant install ruby-2.3.1 (problem with openssl?) (2.3.3 - same) (same without CC=clang)
[CC=clang rvm install ruby-2.3.1 --debug](https://pastebin.com/hQbLpPFg) 
[/home/bjorn/.rvm/log/1493399450_ruby-2.3.1/make.log](https://pastebin.com/hNqAa0jX) 

2. rvm reinstall all --force

```
ruby-2.4.0 - #removing src/ruby-2.4.0 - please wait
ruby-2.4.0 - #removing rubies/ruby-2.4.0 - please wait
Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.25/x86_64/ruby-2.4.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for arch.
Requirements installation successful.
Installing Ruby from source to: /home/bjorn/.rvm/rubies/ruby-2.4.0, this may take a while depending on your cpu(s)...
ruby-2.4.0 - #downloading ruby-2.4.0, this may take a while depending on your connection...
ruby-2.4.0 - #extracting ruby-2.4.0 to /home/bjorn/.rvm/src/ruby-2.4.0 - please wait
ruby-2.4.0 - #configuring - please wait
ruby-2.4.0 - #post-configuration - please wait
ruby-2.4.0 - #compiling - please wait
ruby-2.4.0 - #installing - please wait
ruby-2.4.0 - #making binaries executable - please wait
ruby-2.4.0 - #downloading rubygems-2.6.11
ruby-2.4.0 - #extracting rubygems-2.6.11 - please wait
ruby-2.4.0 - #removing old rubygems - please wait
ruby-2.4.0 - #installing rubygems-2.6.11 - please wait
ruby-2.4.0 - #gemset created /home/bjorn/.rvm/gems/ruby-2.4.0@global
ruby-2.4.0 - #importing gemset /home/bjorn/.rvm/gemsets/global.gems - please wait
ruby-2.4.0 - #generating global wrappers - please wait
ruby-2.4.0 - #gemset created /home/bjorn/.rvm/gems/ruby-2.4.0
ruby-2.4.0 - #importing gemsetfile /home/bjorn/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-2.4.0 - #generating default wrappers - please wait
ruby-2.4.0 - #adjusting #shebangs for (gem irb erb ri rdoc testrb rake).
Install of ruby-2.4.0 - #complete 
Ruby was built without documentation, to build it run: rvm docs generate-ri
Making gemset ruby-2.4.0 pristine - please wait
Making gemset ruby-2.4.0@global pristine - please wait
Unknown ruby interpreter version (do not know how to handle): all.

```

OS - archlinux, tried reinstall with `rvm implode` 
[env|sort](https://pastebin.com/K3gMu2fZ)

**Comment by mpapis** [comment]
it looks like ruby-openssl problem, please check other open issues if they contain any information about 2.3 and backporting openssl patch from ruby 2.4

**Comment by BjornMelgaard** [comment]
[solution for archlinux](https://github.com/rvm/rvm/issues/3958#issuecomment-297342512)

**Comment by mpapis** [comment]
@BjornMelgaard this is not proper solution, we (@rvm) need to add a patch that will fix installation with system openssl, the problem is finding proper patch that from MRI.

**Comment by pkuczynski** [comment]
@BjornMelgaard, I prepared a fix for you. Could you please try following:

```
rvm remove 2.3.1
rvm get branch /bugfix/4006-openssl-arch
rvm install 2.3.1
```

Once this is a confirmed fix, I will update https://wiki.archlinux.org/index.php/RVM

**Comment by BjornMelgaard** [comment]
I can rvm implode and install again
```
$ rvm install 2.3.1
Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.25/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for arch.
Removing undesired packages: openssl - please wait
Error running 'requirements_arch_libs_remove openssl',
please read /home/bjorn/.rvm/log/1494931808_ruby-2.3.1/package_remove_openssl.log
Requirements installation failed with status: 1.
$ cat ~/.rvm/log/1494931808_ruby-2.3.1/package_remove_openssl.log
+requirements_arch_libs_remove:2> __rvm_try_sudo pacman -Rs --noconfirm openssl
+__rvm_try_sudo:2> typeset -a command_to_run
+__rvm_try_sudo:3> typeset sudo_path sbin_path missing_paths
+__rvm_try_sudo:4> command_to_run=( pacman -Rs --noconfirm openssl )
+__rvm_try_sudo:5> ((  UID == 0  ))
+__rvm_try_sudo:6> case 4 (0)
+__rvm_try_sudo:6> case 4 (1)
+__rvm_try_sudo:6> case 4 (2)
+__rvm_try_sudo:6> case 4 (*)
+__rvm_try_sudo:21> is_a_function __rvm_sudo
+is_a_function:2> typeset -f __rvm_sudo
+__rvm_try_sudo:23> missing_paths=''
+__rvm_try_sudo:24> sbin_path=/sbin
+__rvm_try_sudo:26> [[ -d /sbin ]]
+__rvm_try_sudo:26> [[ :/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/home/bjorn/.rvm/bin:/home/bjorn/.node_modules/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl: != *:/sbin:* ]]
+__rvm_try_sudo:27> missing_paths+=:/sbin
+__rvm_try_sudo:24> sbin_path=/usr/sbin
+__rvm_try_sudo:26> [[ -d /usr/sbin ]]
+__rvm_try_sudo:26> [[ :/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/home/bjorn/.rvm/bin:/home/bjorn/.node_modules/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl: != *:/usr/sbin:* ]]
+__rvm_try_sudo:27> missing_paths+=:/usr/sbin
+__rvm_try_sudo:24> sbin_path=/usr/local/sbin
+__rvm_try_sudo:26> [[ -d /usr/local/sbin ]]
+__rvm_try_sudo:26> [[ :/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/home/bjorn/.rvm/bin:/home/bjorn/.node_modules/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl: != *:/usr/local/sbin:* ]]
+__rvm_try_sudo:30> [[ -n :/sbin:/usr/sbin ]]
+__rvm_try_sudo:31> command_to_run=( /usr/bin/env 'PATH=/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/home/bjorn/.rvm/bin:/home/bjorn/.node_modules/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/sbin:/usr/sbin' pacman -Rs --noconfirm openssl )
+__rvm_try_sudo:33> command_to_run=( __rvm_sudo -p '%p password required for '\''pacman -Rs --noconfirm openssl'\'': ' /usr/bin/env 'PATH=/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/home/bjorn/.rvm/bin:/home/bjorn/.node_modules/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/sbin:/usr/sbin' pacman -Rs --noconfirm openssl )
+__rvm_try_sudo:40> __rvm_sudo -p '%p password required for '\''pacman -Rs --noconfirm openssl'\'': ' /usr/bin/env 'PATH=/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/home/bjorn/.rvm/bin:/home/bjorn/.node_modules/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/sbin:/usr/sbin' pacman -Rs --noconfirm openssl
+__rvm_sudo:2> sudo -p '%p password required for '\''pacman -Rs --noconfirm openssl'\'': ' /usr/bin/env 'PATH=/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/home/bjorn/.rvm/bin:/home/bjorn/.node_modules/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/sbin:/usr/sbin' pacman -Rs --noconfirm openssl
checking dependencies...
error: failed to prepare transaction (could not satisfy dependencies)
:: anki20-bin: removing openssl breaks dependency 'openssl'
:: bind-tools: removing openssl breaks dependency 'openssl'
:: coreutils: removing openssl breaks dependency 'openssl'
:: curl: removing openssl breaks dependency 'openssl'
:: elinks: removing openssl breaks dependency 'openssl'
:: git: removing openssl breaks dependency 'openssl'
:: hostapd: removing openssl breaks dependency 'openssl'
:: iputils: removing openssl breaks dependency 'openssl'
:: irssi: removing openssl breaks dependency 'openssl'
:: ldns: removing openssl breaks dependency 'openssl'
:: libarchive: removing openssl breaks dependency 'openssl'
:: libcurl-compat: removing openssl breaks dependency 'openssl'
:: libevent: removing openssl breaks dependency 'openssl'
:: libmariadbclient: removing openssl breaks dependency 'openssl'
:: libsasl: removing openssl breaks dependency 'openssl'
:: libshout: removing openssl breaks dependency 'openssl'
:: libssh2: removing openssl breaks dependency 'openssl'
:: libvncserver: removing openssl breaks dependency 'openssl'
:: lynx: removing openssl breaks dependency 'openssl'
:: nmap: removing openssl breaks dependency 'openssl'
:: ntp: removing openssl breaks dependency 'openssl'
:: openssh: removing openssl breaks dependency 'openssl'
:: openvpn: removing openssl breaks dependency 'openssl'
:: pkcs11-helper: removing openssl breaks dependency 'openssl'
:: postgresql: removing openssl breaks dependency 'openssl>=1.0.0'
:: postgresql-libs: removing openssl breaks dependency 'openssl>=1.0.0'
:: python: removing openssl breaks dependency 'openssl'
:: python2: removing openssl breaks dependency 'openssl'
:: ruby: removing openssl breaks dependency 'openssl'
:: s-nail: removing openssl breaks dependency 'openssl'
:: tcpdump: removing openssl breaks dependency 'openssl'
:: vpnc: removing openssl breaks dependency 'openssl'
:: w3m: removing openssl breaks dependency 'openssl'
:: wpa_supplicant: removing openssl breaks dependency 'openssl'
:: x11vnc: removing openssl breaks dependency 'openssl'
+__rvm_try_sudo:40> return 1
+requirements_arch_libs_remove:2> return 1

```

**Comment by pkuczynski** [comment]
@BjornMelgaard, could you please try again:

```
rvm get branch /bugfix/4006-openssl-arch
rvm install 2.3.1
```

**Comment by BjornMelgaard** [comment]
@pkuczynski  I made implode and ... 

```
 ~  rvm get branch /bugfix/4006-openssl-arch
rvm install 2.3.1

Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/wayneeseguin/rvm/master/binscripts/rvm-installer.asc
Verifying /home/bjorn/.rvm/archives/rvm-installer.asc
gpg: Signature made Mon 20 Feb 2017 04:05:42 PM EET
gpg:                using RSA key 62C9E5F4DA300D94AC36166BE206C29FBF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>" [unknown]
gpg: Note: This key has expired!
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/home/bjorn/.rvm/archives/rvm-installer'
Downloading https://github.com/rvm/rvm/archive/bugfix/4006-openssl-arch.tar.gz

Upgrading the RVM installation in /home/bjorn/.rvm/
    RVM PATH line found in /home/bjorn/.mkshrc /home/bjorn/.profile /home/bjorn/.bashrc /home/bjorn/.zshrc.
    RVM sourcing line found in /home/bjorn/.profile /home/bjorn/.bash_profile /home/bjorn/.zlogin.
Upgrade of RVM in /home/bjorn/.rvm/ is complete.

# BjornMelgaard,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: https://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:

  * No new notes to display.

RVM reloaded!
Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.25/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for arch.
Requirements installation successful.
Installing Ruby from source to: /home/bjorn/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.7M  100 13.7M    0     0   540k      0  0:00:26  0:00:26 --:--:--  683k
ruby-2.3.1 - #extracting ruby-2.3.1 to /home/bjorn/.rvm/src/ruby-2.3.1....
ruby-2.3.1 - #configuring..........................................................
ruby-2.3.1 - #post-configuration..
ruby-2.3.1 - #compiling.................................................................................................
Error running '__rvm_make -j4',
please read /home/bjorn/.rvm/log/1494935495_ruby-2.3.1/make.log
There has been an error while running make. Halting the installation.
```

```
 ~  tail -n 100 ~/.rvm/log/1494935495_ruby-2.3.1/make.log 
      ^
ossl_x509revoked.c: In function ‘ossl_x509revoked_get_serial’:
In file included from /usr/include/openssl/x509.h:23:0,
                 from /usr/include/openssl/x509v3.h:14,
                 from ossl.h:58,
                 from ossl_pkey_dh.c:12:
openssl_missing.h:82:6: error: expected declaration specifiers or ‘...’ before ‘(’ token
 void EVP_MD_CTX_init(EVP_MD_CTX *ctx);
      ^
openssl_missing.h:90:6: error: expected declaration specifiers or ‘...’ before ‘(’ token
 void EVP_MD_CTX_destroy(EVP_MD_CTX *ctx);
      ^
ossl_x509revoked.c:119:34: error: dereferencing pointer to incomplete type ‘X509_REVOKED {aka struct x509_revoked_st}’
     return asn1integer_to_num(rev->serialNumber);
                                  ^~
ossl_pkey_dh.c: In function ‘ossl_dh_new’:
ossl_pkey_dh.c:70:24: error: dereferencing pointer to incomplete type ‘EVP_PKEY {aka struct evp_pkey_st}’
  if (EVP_PKEY_type(pkey->type) != EVP_PKEY_DH) {
                        ^~
ossl_pkey_dh.c: In function ‘dh_generate’:
ossl_pkey_dh.c:107:14: error: storage size of ‘cb’ isn’t known
     BN_GENCB cb;
              ^~
ossl_pkey_dh.c:107:14: warning: unused variable ‘cb’ [-Wunused-variable]
ossl_x509revoked.c: In function ‘ossl_x509revoked_get_time’:
ossl_x509revoked.c:141:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
ossl_x509revoked.c: In function ‘ossl_x509revoked_get_serial’:
ossl_x509revoked.c:120:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[2]: *** [Makefile:301: ossl_x509revoked.o] Error 1
ossl_pkey_dh.c: In function ‘ossl_dh_is_private’:
ossl_pkey_dh.c:278:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
ossl_pkey_dh.c: In function ‘ossl_dh_is_public’:
ossl_pkey_dh.c:261:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[2]: *** [Makefile:301: ossl_pkey_dh.o] Error 1
In file included from ossl.h:221:0,
                 from ossl_pkey.c:10:
openssl_missing.h:78:35: error: macro "EVP_MD_CTX_create" passed 1 arguments, but takes just 0
 EVP_MD_CTX *EVP_MD_CTX_create(void);
                                   ^
In file included from /usr/include/openssl/x509.h:23:0,
                 from /usr/include/openssl/x509v3.h:14,
                 from ossl.h:58,
                 from ossl_pkey.c:10:
openssl_missing.h:82:6: error: expected declaration specifiers or ‘...’ before ‘(’ token
 void EVP_MD_CTX_init(EVP_MD_CTX *ctx);
      ^
openssl_missing.h:90:6: error: expected declaration specifiers or ‘...’ before ‘(’ token
 void EVP_MD_CTX_destroy(EVP_MD_CTX *ctx);
      ^
ossl_pkey.c: In function ‘ossl_generate_cb_2’:
ossl_pkey.c:44:44: error: dereferencing pointer to incomplete type ‘BN_GENCB {aka struct bn_gencb_st}’
     arg = (struct ossl_generate_cb_arg *)cb->arg;
                                            ^~
ossl_pkey.c: In function ‘ossl_pkey_new’:
ossl_pkey.c:94:31: error: dereferencing pointer to incomplete type ‘EVP_PKEY {aka struct evp_pkey_st}’
     switch (EVP_PKEY_type(pkey->type)) {
                               ^~
ossl_pkey.c: In function ‘DupPKeyPtr’:
ossl_pkey.c:215:5: warning: implicit declaration of function ‘CRYPTO_add’ [-Wimplicit-function-declaration]
     CRYPTO_add(&pkey->references, 1, CRYPTO_LOCK_EVP_PKEY);
     ^~~~~~~~~~
ossl_pkey.c:215:38: error: ‘CRYPTO_LOCK_EVP_PKEY’ undeclared (first use in this function)
     CRYPTO_add(&pkey->references, 1, CRYPTO_LOCK_EVP_PKEY);
                                      ^~~~~~~~~~~~~~~~~~~~
ossl_pkey.c:215:38: note: each undeclared identifier is reported only once for each function it appears in
ossl_pkey.c: In function ‘DupPrivPKeyPtr’:
ossl_pkey.c:229:38: error: ‘CRYPTO_LOCK_EVP_PKEY’ undeclared (first use in this function)
     CRYPTO_add(&pkey->references, 1, CRYPTO_LOCK_EVP_PKEY);
                                      ^~~~~~~~~~~~~~~~~~~~
ossl_pkey.c: In function ‘ossl_pkey_sign’:
ossl_pkey.c:289:16: error: storage size of ‘ctx’ isn’t known
     EVP_MD_CTX ctx;
                ^~~
ossl_pkey.c:289:16: warning: unused variable ‘ctx’ [-Wunused-variable]
ossl_pkey.c: In function ‘ossl_pkey_verify’:
ossl_pkey.c:337:16: error: storage size of ‘ctx’ isn’t known
     EVP_MD_CTX ctx;
                ^~~
ossl_pkey.c:337:16: warning: unused variable ‘ctx’ [-Wunused-variable]
make[2]: *** [Makefile:301: ossl_pkey.o] Error 1
make[2]: Leaving directory '/home/bjorn/.rvm/src/ruby-2.3.1/ext/openssl'
make[1]: *** [exts.mk:209: ext/openssl/all] Error 2
make[1]: *** Waiting for unfinished jobs....
linking shared-object json/ext/generator.so
make[2]: Leaving directory '/home/bjorn/.rvm/src/ruby-2.3.1/ext/json/generator'
linking shared-object objspace.so
make[2]: Leaving directory '/home/bjorn/.rvm/src/ruby-2.3.1/ext/objspace'
linking shared-object nkf.so
make[2]: Leaving directory '/home/bjorn/.rvm/src/ruby-2.3.1/ext/nkf'
make[1]: Leaving directory '/home/bjorn/.rvm/src/ruby-2.3.1'
make: *** [uncommon.mk:203: build-ext] Error 2
++ return 2
```

also it seems like you rootified my dirs)
```
$ gem install bundler neovim 
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /usr/lib/ruby/gems/2.4.0 directory.
```

**Comment by pkuczynski** [comment]
The only change which was made in this branch was about openssl library. I know old versions of ruby are not compatible with latest openssl 1.1. Ruby does not plan to support new openssl with old ruby. So I need to figure out an alternative solution. Let me think about it...

**Comment by pkuczynski** [comment]
@BjornMelgaard can you check if you have now openssl-1.0 installed in your system and what's the location for it?

**Comment by BjornMelgaard** [comment]
```
 ~  whereis openssl
openssl: /usr/bin/openssl /usr/include/openssl /usr/share/man/man1/openssl.1ssl.gz
 ~  pacman -Qi openssl
Name            : openssl
Version         : 1.1.0.e-1
Description     : The Open Source toolkit for Secure Sockets Layer and Transport Layer Security
Architecture    : x86_64
URL             : https://www.openssl.org
Licenses        : custom:BSD
Groups          : None
Provides        : None
Depends On      : perl
Optional Deps   : ca-certificates [installed]
Required By     : anki20-bin  bind-tools  coreutils  curl  elinks  git  hostapd  iputils  irssi  ldns  libarchive  libcurl-compat
                  libevent  libmariadbclient  libsasl  libshout  libssh2  libvncserver  lynx  nmap  ntp  openssh  openvpn
                  pkcs11-helper  postgresql  postgresql-libs  python  python2  ruby  s-nail  tcpdump  vpnc  w3m  wpa_supplicant  x11vnc
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 6.23 MiB
Packager        : Pierre Schmitz <pierre@archlinux.de>
Build Date      : Thu 16 Feb 2017 04:53:19 PM EET
Install Date    : Tue 16 May 2017 09:20:03 AM EEST
Install Reason  : Installed as a dependency for another package
Install Script  : No
Validated By    : Signature


```

**Comment by pkuczynski** [comment]
@BjornMelgaard and now please:

```
ls -al /usr/lib/openssl-1.0/
```

**Comment by BjornMelgaard** [comment]
```
 ~  ls -al /usr/lib/openssl-1.0/
total 168
drwxr-xr-x   4 root root   4096 Apr  5 10:12 .
drwxr-xr-x 215 root root 155648 May 16 09:20 ..
drwxr-xr-x   2 root root   4096 Apr  5 10:12 engines
lrwxrwxrwx   1 root root     21 Apr  5 10:12 libcrypto.so -> ../libcrypto.so.1.0.0
lrwxrwxrwx   1 root root     18 Apr  5 10:12 libssl.so -> ../libssl.so.1.0.0
drwxr-xr-x   2 root root   4096 Apr  5 10:12 pkgconfig
```

**Comment by pkuczynski** [comment]
One more time please :)

```
rvm get branch /bugfix/4006-openssl-arch
rvm install 2.3.1
```

**Comment by pkuczynski** [comment]
@BjornMelgaard any luck this time?

**Comment by BjornMelgaard** [comment]
Sorry, I forgot :smile: 

```
 ~  rvm get branch /bugfix/4006-openssl-arch
Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/wayneeseguin/rvm/master/binscripts/rvm-installer.asc
Verifying /home/bjorn/.rvm/archives/rvm-installer.asc
gpg: Signature made Mon 20 Feb 2017 04:05:42 PM EET
gpg:                using RSA key 62C9E5F4DA300D94AC36166BE206C29FBF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>" [unknown]
gpg: Note: This key has expired!
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/home/bjorn/.rvm/archives/rvm-installer'
Downloading https://github.com/rvm/rvm/archive/bugfix/4006-openssl-arch.tar.gz

Upgrading the RVM installation in /home/bjorn/.rvm/
    RVM PATH line found in /home/bjorn/.mkshrc /home/bjorn/.profile /home/bjorn/.bashrc /home/bjorn/.zshrc.
    RVM sourcing line found in /home/bjorn/.profile /home/bjorn/.bash_profile /home/bjorn/.zshrc /home/bjorn/.zlogin.
Upgrade of RVM in /home/bjorn/.rvm/ is complete.

# BjornMelgaard,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: https://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:

  * No new notes to display.

RVM reloaded!
 ~  rvm install 2.3.1
Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.25/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for arch.
Requirements installation successful.
Installing Ruby from source to: /home/bjorn/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.7M  100 13.7M    0     0   372k      0  0:00:37  0:00:37 --:--:--  429k
ruby-2.3.1 - #extracting ruby-2.3.1 to /home/bjorn/.rvm/src/ruby-2.3.1 - please wait
ruby-2.3.1 - #configuring - please wait
ruby-2.3.1 - #post-configuration - please wait
ruby-2.3.1 - #compiling - please wait
Error running '__rvm_make -j4',
please read /home/bjorn/.rvm/log/1495140357_ruby-2.3.1/make.log
There has been an error while running make. Halting the installation.

```
[pastebin](https://pastebin.com/PgT5f0tU)

**Comment by pkuczynski** [comment]
Would need your test again please:

```
rvm get branch /bugfix/4006-openssl-arch
rvm install 2.3.1
```

**Comment by BjornMelgaard** [comment]
of course ) 

I run `rvm get branch /bugfix/4006-openssl-arch`, found that there error with `PATH not prop...`, made `rvm implode`, install with dotfiles (im using zsh)

```
#.zlogin
[[ -s "$HOME/.rvm/scripts/rvm" ]] && source "$HOME/.rvm/scripts/rvm" # Load RVM into a shell session *as a function*
#.zshrc
export PATH="$PATH:$HOME/.rvm/bin"
```

```
 ~  rvm get branch /bugfix/4006-openssl-arch
rvm install 2.3.1
Warning! PATH is not properly set up, '/home/bjorn/.rvm/gems/ruby-2.4.0/bin' is not at first place.
         Usually this is caused by shell initialization files. Search for 'PATH=...' entries.
         You can also re-add RVM to your profile by running: 'rvm get stable --auto-dotfiles'.
         To fix it temporarily in this shell session run: 'rvm use ruby-2.4.0'.
         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.
Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/wayneeseguin/rvm/master/binscripts/rvm-installer.asc
Verifying /home/bjorn/.rvm/archives/rvm-installer.asc
gpg: Signature made Mon 20 Feb 2017 04:05:42 PM EET
gpg:                using RSA key 62C9E5F4DA300D94AC36166BE206C29FBF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>" [unknown]
gpg: Note: This key has expired!
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/home/bjorn/.rvm/archives/rvm-installer'
Downloading https://github.com/rvm/rvm/archive/bugfix/4006-openssl-arch.tar.gz

Upgrading the RVM installation in /home/bjorn/.rvm/
    RVM PATH line found in /home/bjorn/.mkshrc /home/bjorn/.profile /home/bjorn/.bashrc /home/bjorn/.zshrc.
    RVM sourcing line found in /home/bjorn/.profile /home/bjorn/.bash_profile /home/bjorn/.zlogin.
Upgrade of RVM in /home/bjorn/.rvm/ is complete.

# BjornMelgaard,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: https://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:

  * No new notes to display.

RVM reloaded!
Warning! PATH is not properly set up, '/home/bjorn/.rvm/gems/ruby-2.4.0/bin' is not at first place.
         Usually this is caused by shell initialization files. Search for 'PATH=...' entries.
         You can also re-add RVM to your profile by running: 'rvm get stable --auto-dotfiles'.
         To fix it temporarily in this shell session run: 'rvm use ruby-2.4.0'.
         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.
Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.25/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for arch.
Requirements installation successful.
Installing Ruby from source to: /home/bjorn/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.7M  100 13.7M    0     0  98046      0  0:02:27  0:02:27 --:--:-- 94214
ruby-2.3.1 - #extracting ruby-2.3.1 to /home/bjorn/.rvm/src/ruby-2.3.1....
ruby-2.3.1 - #configuring.
Error running './configure --prefix=/home/bjorn/.rvm/rubies/ruby-2.3.1 --with-openssl-dir/usr/lib/openssl-1.0 --disable-install-doc --enable-shared',
please read /home/bjorn/.rvm/log/1495203762_ruby-2.3.1/configure.log
There has been an error while running configure. Halting the installation.
 ✘  ~  cat /home/bjorn/.rvm/log/1495203762_ruby-2.3.1/configure.log
[2017-05-19 17:22:45] ./configure
current path: /home/bjorn/.rvm/src/ruby-2.3.1
GEM_HOME=/home/bjorn/.rvm/gems/ruby-2.4.0
GEM_PATH=/home/bjorn/.rvm/gems/ruby-2.4.0:/home/bjorn/.rvm/gems/ruby-2.4.0@global
PATH=/home/bjorn/.node_modules/bin:/home/bjorn/.cabal/bin:/usr/local/heroku/bin:/home/bjorn/.bin:/home/bjorn/projects/umsf/bin:/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/bjorn/.rvm/bin:/home/bjorn/.rvm/bin
command(5): ./configure --prefix=/home/bjorn/.rvm/rubies/ruby-2.3.1 --with-openssl-dir/usr/lib/openssl-1.0 --disable-install-doc --enable-shared
configure: error: invalid package name: openssl-dir/usr/lib/openssl-1.0
```
and now I cant fix  `PATH is not properly set up` :dizzy_face: 
```
 ~  rvm list                                                   
Warning! PATH is not properly set up, '/home/bjorn/.rvm/gems/ruby-2.4.0/bin' is not at first place.
         Usually this is caused by shell initialization files. Search for 'PATH=...' entries.
         You can also re-add RVM to your profile by running: 'rvm get stable --auto-dotfiles'.
         To fix it temporarily in this shell session run: 'rvm use ruby-2.4.0'.
         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.

rvm rubies


# No rvm rubies installed yet. Try 'rvm help install'.

 ~  echo $PATH | tr ':' '\n'    
/home/bjorn/.node_modules/bin
/home/bjorn/.cabal/bin
/usr/local/heroku/bin
/home/bjorn/.bin
/home/bjorn/projects/umsf/bin
/home/bjorn/.rvm/gems/ruby-2.4.0/bin
/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin
/home/bjorn/.rvm/rubies/ruby-2.4.0/bin
/usr/local/sbin
/usr/local/bin
/usr/bin
/usr/lib/jvm/default/bin
/usr/bin/site_perl
/usr/bin/vendor_perl
/usr/bin/core_perl
/home/bjorn/.rvm/bin
/home/bjorn/.rvm/bin
 ~  echo $GEM_HOME              
/home/bjorn/.rvm/gems/ruby-2.4.0

```

**Comment by BjornMelgaard** [comment]
Fixed :scream:

**Comment by pkuczynski** [comment]
Aaaaaaaaa, what a stupid typo! Apologies. Can you try again, please?

```
rvm get branch /bugfix/4006-openssl-arch
rvm install 2.3.1
```

**Comment by BjornMelgaard** [comment]
Nope) [tail (sorry, hasnt saved another)](https://pastebin.com/0UuxjpSe)

**Comment by pkuczynski** [comment]
The tail is not enough as I would need to see how rvm is calling compilation. Can you please post the full log (incl. configure and make.log) from:

```
rvm install 2.3.1 --debug
```

**Comment by BjornMelgaard** [comment]
```
✘  ~  rvm install 2.3.1 --debug
ruby-2.3.1 - install
Log prefix: /home/bjorn/.rvm/log/1495207943_ruby-2.3.1/
Log filesystem: ext4
Log max name length: 250
Log file: /home/bjorn/.rvm/log/1495207943_ruby-2.3.1/remove.src.log
[2017-05-19 18:32:23] __rvm_rm_rf
__rvm_rm_rf () {
	__rvm_rm_rf_verbose "$@"
}
current path: /home/bjorn
GEM_PATH=/home/bjorn/.rvm/gems/ruby-2.4.1:/home/bjorn/.rvm/gems/ruby-2.4.1@global
GEM_HOME=/home/bjorn/.rvm/gems/ruby-2.4.1
PATH=/home/bjorn/.rvm/gems/ruby-2.4.1/bin:/home/bjorn/.rvm/gems/ruby-2.4.1@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.1/bin:/home/bjorn/.node_modules/bin:/home/bjorn/.cabal/bin:/usr/local/heroku/bin:/home/bjorn/.bin:/home/bjorn/projects/umsf/bin:/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/bjorn/.rvm/bin:/home/bjorn/.rvm/bin
command(2): __rvm_rm_rf /home/bjorn/.rvm/src/ruby-2.3.1
ruby-2.3.1 - #removing src/ruby-2.3.1 - please wait
ruby-2.3.1 - #already removed rubies/ruby-2.3.1
Searching for binary rubies, this might take some time.
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.3.1.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.3.1.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rubies.travis-ci.org/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2 --insecure
Remote file does not exist https://rubies.travis-ci.org/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2
rvm_remote_server_url3 not found
No remote file name found
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.3.1.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.3.1.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rubies.travis-ci.org/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2 --insecure
Remote file does not exist https://rubies.travis-ci.org/arch/libc-2.25/x86_64/ruby-2.3.1.tar.bz2
rvm_remote_server_url3 not found
No remote file name found
No binary rubies available for: arch/libc-2.25/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Free disk space 52708MB, required 440MB.
__rvm_setup_compile_environment_setup ruby-2.3.1
rvm_autolibs_flag=4
__rvm_setup_compile_environment_movable_early ruby-2.3.1
__rvm_setup_compile_environment_system_early ruby-2.3.1
__rvm_setup_compile_environment_requirements ruby-2.3.1
Checking requirements for arch.
requirements code for arch loaded
rvm_configure_flags+=( --with-openssl-dir=/usr/lib/openssl-1.0 )
Found required packages: bash, curl, patch, gcc, patch, curl, zlib, readline, autoconf, automake, diffutils, make, libtool, bison, sqlite, git, openssl-1.0.
Requirements installation successful.
__rvm_setup_compile_environment_movable ruby-2.3.1
__rvm_setup_compile_environment_parse_name ruby-2.3.1
__rvm_setup_compile_environment_system ruby-2.3.1
__rvm_setup_compile_environment_architectures ruby-2.3.1
rvm_architectures(0):.
__rvm_setup_compile_environment_gcc47plus ruby-2.3.1
__rvm_setup_compile_environment_bison ruby-2.3.1
__rvm_setup_compile_environment_flags_docs ruby-2.3.1
__rvm_setup_compile_environment_flags_shared_static ruby-2.3.1
__rvm_setup_compile_environment_flags_threads ruby-2.3.1
__rvm_setup_compile_environment_compatibility_flag ruby-2.3.1
found compiler: /usr/bin/gcc
Installing Ruby from source to: /home/bjorn/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
Load checksums for https://cache.ruby-lang.org/pub/ruby/2.3/ruby-2.3.1.tar.bz2
Fetch https://cache.ruby-lang.org/pub/ruby/2.3/ruby-2.3.1.tar.bz2 => ruby-2.3.1.tar.bz2
Validating checksums for file /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2
    ...checksums found in db
Validating md5 checksum
Calculate md5 checksum for /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2
    ...md5 checksum is valid!
Validating sha15 checksum
Calculate sha512 checksum for 512 /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2
    ...sha512 checksum is valid!
    ...archive checksum matched, not downloading
Validating checksums for file /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2
    ...checksums found in db
Validating md5 checksum
Calculate md5 checksum for /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2
    ...md5 checksum is valid!
Validating sha15 checksum
Calculate sha512 checksum for 512 /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2
    ...sha512 checksum is valid!
Downloaded archive checksum matched.
Log file: /home/bjorn/.rvm/log/1495207943_ruby-2.3.1/extract.log
[2017-05-19 18:32:27] __rvm_package_extract
__rvm_package_extract () {
	rvm_debug __rvm_package_extract:$#: "$@"
	\typeset __tempdir __path __file __return
	__return=0 
	__tempdir="$( TMPDIR="${rvm_tmp_path}" mktemp -d -t rvm-tmp.XXXXXXXXX )" 
	__rvm_package_extract_run "$1" "$__tempdir" || __return=$? 
	if (( __return == 0 ))
	then
		for __path in "$__tempdir"/*
		do
			__file="${__path##*/}" 
			if [[ -n "${__file}" && -e "$2/${__file}" ]]
			then
				\command \rm -rf "$2/${__file}" || __return=$? 
			fi
			\command \mv -f "${__path}" "$2/" || __return=$? 
		done
	fi
	if [[ -n "$__tempdir" ]]
	then
		\command \rm -rf "$__tempdir"
	fi
	return $__return
}
current path: /home/bjorn/.rvm/src
GEM_PATH=/home/bjorn/.rvm/gems/ruby-2.4.1:/home/bjorn/.rvm/gems/ruby-2.4.1@global
GEM_HOME=/home/bjorn/.rvm/gems/ruby-2.4.1
PATH=/home/bjorn/.rvm/gems/ruby-2.4.1/bin:/home/bjorn/.rvm/gems/ruby-2.4.1@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.1/bin:/home/bjorn/.node_modules/bin:/home/bjorn/.cabal/bin:/usr/local/heroku/bin:/home/bjorn/.bin:/home/bjorn/projects/umsf/bin:/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/bjorn/.rvm/bin:/home/bjorn/.rvm/bin
command(3): __rvm_package_extract /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2 /home/bjorn/.rvm/tmp/rvm_src_9191
ruby-2.3.1 - #extracting ruby-2.3.1 to /home/bjorn/.rvm/src/ruby-2.3.1 - please wait
__rvm_package_extract:2: /home/bjorn/.rvm/archives/ruby-2.3.1.tar.bz2 /home/bjorn/.rvm/tmp/rvm_src_9191
__rvm_rm_rf already gone: /home/bjorn/.rvm/src/ruby-2.3.1
Patch name 'optional', expanded_name(s) ''.
Patch name 'default', expanded_name(s) ''.
Patch name 'arch', expanded_name(s) ''.
All found patches(0): .
Log file: /home/bjorn/.rvm/log/1495207943_ruby-2.3.1/configure.log
[2017-05-19 18:32:31] ./configure
current path: /home/bjorn/.rvm/src/ruby-2.3.1
GEM_PATH=/home/bjorn/.rvm/gems/ruby-2.4.1:/home/bjorn/.rvm/gems/ruby-2.4.1@global
GEM_HOME=/home/bjorn/.rvm/gems/ruby-2.4.1
PATH=/home/bjorn/.rvm/gems/ruby-2.4.1/bin:/home/bjorn/.rvm/gems/ruby-2.4.1@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.1/bin:/home/bjorn/.node_modules/bin:/home/bjorn/.cabal/bin:/usr/local/heroku/bin:/home/bjorn/.bin:/home/bjorn/projects/umsf/bin:/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/bjorn/.rvm/bin:/home/bjorn/.rvm/bin
command(6): ./configure --prefix=/home/bjorn/.rvm/rubies/ruby-2.3.1  --with-openssl-dir=/usr/lib/openssl-1.0 --disable-install-doc --enable-shared
ruby-2.3.1 - #configuring - please wait
Log file: /home/bjorn/.rvm/log/1495207943_ruby-2.3.1/postconfigure.log
[2017-05-19 18:33:02] __rvm_post_configure_ruby
__rvm_post_configure_ruby () {
	__rvm_post_configure_ruby_fix_athlon_sse2
	__rvm_post_configure_ruby_update_setup
}
current path: /home/bjorn/.rvm/src/ruby-2.3.1
GEM_PATH=/home/bjorn/.rvm/gems/ruby-2.4.1:/home/bjorn/.rvm/gems/ruby-2.4.1@global
GEM_HOME=/home/bjorn/.rvm/gems/ruby-2.4.1
PATH=/home/bjorn/.rvm/gems/ruby-2.4.1/bin:/home/bjorn/.rvm/gems/ruby-2.4.1@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.1/bin:/home/bjorn/.node_modules/bin:/home/bjorn/.cabal/bin:/usr/local/heroku/bin:/home/bjorn/.bin:/home/bjorn/projects/umsf/bin:/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/bjorn/.rvm/bin:/home/bjorn/.rvm/bin
command(1): __rvm_post_configure_ruby
ruby-2.3.1 - #post-configuration - please wait
Log file: /home/bjorn/.rvm/log/1495207943_ruby-2.3.1/make.log
[2017-05-19 18:33:02] __rvm_make
__rvm_make () {
	\make "$@" || return $?
}
current path: /home/bjorn/.rvm/src/ruby-2.3.1
GEM_PATH=/home/bjorn/.rvm/gems/ruby-2.4.1:/home/bjorn/.rvm/gems/ruby-2.4.1@global
GEM_HOME=/home/bjorn/.rvm/gems/ruby-2.4.1
PATH=/home/bjorn/.rvm/gems/ruby-2.4.1/bin:/home/bjorn/.rvm/gems/ruby-2.4.1@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.1/bin:/home/bjorn/.node_modules/bin:/home/bjorn/.cabal/bin:/usr/local/heroku/bin:/home/bjorn/.bin:/home/bjorn/projects/umsf/bin:/home/bjorn/.rvm/gems/ruby-2.4.0/bin:/home/bjorn/.rvm/gems/ruby-2.4.0@global/bin:/home/bjorn/.rvm/rubies/ruby-2.4.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/bjorn/.rvm/bin:/home/bjorn/.rvm/bin
command(2): __rvm_make -j4
ruby-2.3.1 - #compiling - please wait
Error running '__rvm_make -j4',
please read /home/bjorn/.rvm/log/1495207943_ruby-2.3.1/make.log
There has been an error while running make. Halting the installation.
__rvm_rm_rf already gone: /home/bjorn/.rvm/tmp/9191*

```

https://pastebin.com/raw/uvmQAGCH

**Comment by pkuczynski** [comment]
Thanks @BjornMelgaard. Looking into that...

**Comment by taikhuu** [comment]
I think Arch wiki already had a solution for this ?

```
RVM uses wrong OpenSSL version
Ruby versions older than 2.4 require OpenSSL 1.0 but RVM will try to build them with OpenSSL 1.1. You know this is the case if you find this line in the ~/.rvm/log/XYZ/make.log file:
/usr/include/openssl/asn1_mac.h:10:2: error: #error "This file is obsolete; please update your software."
You can point it to the correct version like this:
$ sudo pacman -S openssl-1.0 # Probably already installed
$ rvm remove <ruby-version>
$ PKG_CONFIG_PATH=/usr/lib/openssl-1.0/pkgconfig:/usr/lib/pkgconfig rvm install <ruby-version>
Alternatively you could also use RVM to install OpenSSL as above:
$ rvm pkg install openssl
$ rvm remove 2.3.4
$ rvm install 2.3.4 -C --with-openssl-dir=$HOME/.rvm/usr

```

**Comment by BjornMelgaard** [comment]
@taikhuu , yes, you are right, I [mentioned this solution](https://github.com/rvm/rvm/issues/4006#issuecomment-301690843)

**Comment by pkuczynski** [comment]
@BjornMelgaard I know the solution from their wiki, but we would like to build it into RVM, so nobody needs to lookup troubleshooting documentation :)

**Comment by BjornMelgaard** [comment]
@pkuczynski Yes, I understand and greatly appreciated )

**Comment by RigDillinger** [comment]
@pkuczynski any updates on this?
Unfortunately, solution from Archwiki didn't work for me

**Comment by andrejcremoznik** [comment]
@RigDillinger could you paste the command here please

**Comment by RigDillinger** [comment]
@andrejcremoznik I installed openssl first: `rvm pkg install openssl`
And then: `rvm install 2.3.1 -C --with-openssl-dir=$HOME/.rvm/usr --with-gcc=clang`

Running without setting `clang` ends up with something like `../../miniruby': corrupted double-linked list` during compilation
But, correct me if i'm wrong, compiling with clang can possibly cause some gems not to build upon this ruby version, at least without some magic.

**Comment by andrejcremoznik** [comment]
> clang can possibly cause some gems not to build upon this ruby version, at least without some magic

Maybe. Installing `bundler` and running `bundle install` fortunately worked fine for my case. Just needed `ruby` running for development on a web project. Needless to say `ruby` isn't my cup of coffee.

**Comment by kruczjak** [comment]
I have similar problem to @RigDillinger only `clang` compiler works but then, I can't install e.g. `unf_ext` gem.

> But, correct me if i'm wrong, compiling with clang can possibly cause some gems not to build upon this ruby version, at least without more magic.

^ looks like that statement is correct :sob: 

```
compiling unf.cc
g++: error: unrecognized command line option '-Wshorten-64-to-32'
g++: error: unrecognized command line option '-Wdivision-by-zero'; did you mean '-Wdiv-by-zero'?
g++: error: unrecognized command line option '-Wextra-tokens'; did you mean '--extra-warnings'?
make: *** [Makefile:207: unf.o] Error 1

make failed, exit code 2
```
`clang version 4.0.0 (tags/RELEASE_400/final)`
`gcc version 7.1.1 20170516 (GCC)`

**Comment by pkuczynski** [comment]
@BjornMelgaard, @RigDillinger, @kruczjak, @andrejcremoznik  could you please try again (you can use different version if you want as long as its lower then 2.4):

```
rvm get branch /bugfix/4006-openssl-arch
rvm remove 2.3.1
rvm install 2.3.1
```

Thanks in advance for your testing!

**Comment by BjornMelgaard** [comment]
```
~  rvm get branch /bugfix/4006-openssl-arch
Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/wayneeseguin/rvm/master/binscripts/rvm-installer.asc
Verifying /home/bjorn/.rvm/archives/rvm-installer.asc
gpg: Signature made Mon 20 Feb 2017 04:05:42 PM EET
gpg:                using RSA key 62C9E5F4DA300D94AC36166BE206C29FBF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>" [unknown]
gpg: Note: This key has expired!
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/home/bjorn/.rvm/archives/rvm-installer'
Downloading https://github.com/rvm/rvm/archive/bugfix/4006-openssl-arch.tar.gz

Upgrading the RVM installation in /home/bjorn/.rvm/
    RVM PATH line found in /home/bjorn/.zshrc.
    RVM PATH line not found for Bash, rerun this command with '--auto-dotfiles' flag to fix it.
    RVM sourcing line found in /home/bjorn/.zshrc.
    RVM sourcing line not found for Bash, rerun this command with '--auto-dotfiles' flag to fix it.
Upgrade of RVM in /home/bjorn/.rvm/ is complete.

# BjornMelgaard,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: https://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:

  * No new notes to display.

RVM reloaded!
 ~  rvm remove 2.3.1
ruby-2.3.1 - #removing src/ruby-2.3.1 - please wait
Using /home/bjorn/.rvm/gems/ruby-2.4.1
 ~  rvm install 2.3.1
Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.25/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for arch.
Requirements installation successful.
Installing Ruby from source to: /home/bjorn/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
ruby-2.3.1 - #extracting ruby-2.3.1 to /home/bjorn/.rvm/src/ruby-2.3.1 - please wait
ruby-2.3.1 - #configuring - please wait
ruby-2.3.1 - #post-configuration - please wait
ruby-2.3.1 - #compiling - please wait
*** Error in `../../miniruby': corrupted double-linked list: 0x0000000001016cc0 ***

```
and there it hanged up

http://dropmefiles.com/Q0aQN

**Comment by RigDillinger** [comment]
@pkuczynski @BjornMelgaard 
> ~  rvm get branch /bugfix/4006-openssl-arch
> ...
> *** Error in `../../miniruby': corrupted double-linked list: 0x0000000001016cc0 *** 
Same

**Comment by pkuczynski** [comment]
@kruczjak so you compiled with `--with-openssl-dir=$HOME/.rvm/usr --with-gcc=clang` or with prepending `PKG_CONFIG_PATH=/usr/lib/openssl-1.0/pkgconfig:/usr/lib/pkgconfig`?

**Comment by nightsh** [comment]
I'm having the same issue as @RigDillinger, should we file a new issue for it, or you guys think it's related?

```
└> rvm install 2.3.3                                     
ruby-2.3.3 - #removing src/ruby-2.3.3 - please wait
Searching for binary rubies, this might take some time.
No binary rubies available for: opensuse/20170617/x86_64/ruby-2.3.3.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for opensuse.
Requirements installation successful.
Installing Ruby from source to: /home/g/.rvm/rubies/ruby-2.3.3, this may take a while depending on your cpu(s)...
ruby-2.3.3 - #downloading ruby-2.3.3, this may take a while depending on your connection...
ruby-2.3.3 - #extracting ruby-2.3.3 to /home/g/.rvm/src/ruby-2.3.3 - please wait
ruby-2.3.3 - #configuring - please wait
ruby-2.3.3 - #post-configuration - please wait
ruby-2.3.3 - #compiling - please wait
*** Error in `../../miniruby': corrupted double-linked list: 0x0000000001e72890 ***
```
It hangs here, does not respond to SIGINT. Had to kill the process.

Environment is 64 bit openSUSE Tumbleweed, up to date.

**Comment by pkuczynski** [comment]
Yes, please create a separate issue for openSUSE. It is most likely connected to this, but solution might be different

**Comment by cars10** [comment]
I am facing similiar issues.

Compiling with `CC=clang rvm install 2.3.3 -C --with-openssl-dir=$HOME/.rvm/usr` works, but alot of gems cannot bundle then: 
```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/cars10/.rvm/gems/ruby-2.3.3@holidaywiki/gems/therubyracer-0.12.1/ext/v8
/home/cars10/.rvm/rubies/ruby-2.3.3/bin/ruby -r ./siteconf20170619-27577-1alviyh.rb extconf.rb
checking for main() in -lpthread... yes
creating Makefile

current directory: /home/cars10/.rvm/gems/ruby-2.3.3@holidaywiki/gems/therubyracer-0.12.1/ext/v8
make "DESTDIR=" clean

current directory: /home/cars10/.rvm/gems/ruby-2.3.3@holidaywiki/gems/therubyracer-0.12.1/ext/v8
make "DESTDIR="
compiling heap.cc
g++: error: unrecognized command line option ‘-Wshorten-64-to-32’
g++: error: unrecognized command line option ‘-Wdivision-by-zero’; did you mean ‘-Wdiv-by-zero’?
g++: error: unrecognized command line option ‘-Wextra-tokens’; did you mean ‘--extra-warnings’?
make: *** [Makefile:207: heap.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/cars10/.rvm/gems/ruby-2.3.3@holidaywiki/gems/therubyracer-0.12.1 for inspection.
Results logged to /home/cars10/.rvm/gems/ruby-2.3.3@holidaywiki/extensions/x86_64-linux/2.3.0/therubyracer-0.12.1/gem_make.out

An error occurred while installing therubyracer (0.12.1), and Bundler cannot continue.
Make sure that `gem install therubyracer -v '0.12.1'` succeeds before bundling.
```


Installing ruby with `PKG_CONFIG_PATH=/usr/lib/openssl-1.0/pkgconfig:/usr/lib/pkgconfig rvm install ruby-2.3.3` leads to the usual `*** Error in `../../miniruby': corrupted double-linked list: 0x00000000024c3130 ***`  error...

**Comment by taikhuu** [comment]
It's another issue. Dowgrade to gcc6 and gcc-libs 6 will let you install ruby

**Comment by RigDillinger** [comment]
@taikhuu thanks for the tip!
For those who are concerned, installing `gcc63-multilib` from AUR and setting `CC=gcc-6.3` allowed to compile `ruby-2.3.1` and build gems upon it.

**Comment by nixon-by** [comment]
I successfully installed ruby 2.3.1 the following way
`sudo pacman -Syyu gcc5`
`CC=gcc-5 rvm install 2.3.1`

**Comment by leizzer** [comment]
@nixon-by and your pacman still works? Be careful with that.

And in my case, after installing it with `--with-openssl-dir=$HOME/.rvm/usr` all my specs fail with open-ssl errors since I need to do some authentication during my tests. I am using Arch and my coworkers with OSx don't have this issue.

@pkuczynski I would gladly do screen share with you if that helps

**Comment by nixon-by** [comment]
@leizzer why should it be crashed? 
I have 2 gcc in the system

so per default i have `gcc 7.1.1 in  /usr/lib/gcc/x86_64-pc-linux-gnu/7.1.1/`
and additionally `gcc 5.4.0 in /usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/`

openssl-1.0 is additionally installed as well in the /usr/lib/openssl-1.0

I suppose both gcc-5 and openssl-1.0 isn't used unless i explicitly specify it somewhere

**Comment by leizzer** [comment]
@nixon-by Nevermind, I misread. I thought you downgraded OpenSSL but you installed the extra package. Last version of pacman uses openssl 1.1 and lot of people, including me, had problems with unsynced mirrors that had openssl 1.0

**Comment by mathieujobin** [comment]
thanks for the solution @nixon-by

**Comment by cars10** [comment]
The solution provided by @nixon-by works. Thank you!

**Comment by renatolond** [comment]
Using the branch by @pkuczynski and openssl-1.0 it's working perfectly. Thanks! :)

**Comment by Corleanus** [comment]
rvm install 2.3.0
ruby-2.3.0 - #removing src/ruby-2.3.0..
Searching for binary rubies, this might take some time.
No binary rubies available for: kali/kali-rolling/x86_64/ruby-2.3.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for kali.
Installing requirements for kali.
Updating system.
Installing required packages: libreadline6-dev....
Requirements installation successful.
Installing Ruby from source to: /usr/local/rvm/rubies/ruby-2.3.0, this may take a while depending on your cpu(s)...
ruby-2.3.0 - #downloading ruby-2.3.0, this may take a while depending on your connection...
ruby-2.3.0 - #extracting ruby-2.3.0 to /usr/local/rvm/src/ruby-2.3.0....
ruby-2.3.0 - #applying patch /usr/local/rvm/patches/ruby/changeset_r53419.diff.
ruby-2.3.0 - #applying patch /usr/local/rvm/patches/ruby/2.3.0/random_c_using_NR_prefix.patch.
ruby-2.3.0 - #configuring......................................................-
ruby-2.3.0 - #post-configuration..
ruby-2.3.0 - #compiling........................................................|
ruby-2.3.0 - #installing..*** Error in `./miniruby': malloc(): memory corruption: 0x00007ffc4762cf98 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x70bfb)[0x7f954735fbfb]
/lib/x86_64-linux-gnu/libc.so.6(+0x76fc6)[0x7f9547365fc6]
/lib/x86_64-linux-gnu/libc.so.6(+0x79089)[0x7f9547368089]
/lib/x86_64-linux-gnu/libc.so.6(__libc_calloc+0x27b)[0x7f954736ab7b]
./miniruby(ruby_xcalloc+0x50)[0x5588c3580200]
./miniruby(st_init_table_with_size+0x4e)[0x5588c363ff9e]
./miniruby(+0xea395)[0x5588c35cc395]
./miniruby(+0xf82b4)[0x5588c35da2b4]
./miniruby(+0x100043)[0x5588c35e2043]
./miniruby(+0x1ee48c)[0x5588c36d048c]
./miniruby(rb_iseq_compile_with_option+0x38d)[0x5588c36aad2d]
./miniruby(+0x1e1517)[0x5588c36c3517]
./miniruby(rb_mod_module_eval+0x165)[0x5588c36c40e5]
./miniruby(+0x1d0f44)[0x5588c36b2f44]
./miniruby(+0x1d886d)[0x5588c36ba86d]
./miniruby(+0x1de8cc)[0x5588c36c08cc]
./miniruby(+0x1df5c7)[0x5588c36c15c7]
./miniruby(rb_yield+0x5f)[0x5588c36c1acf]
./miniruby(rb_ary_each+0x3c)[0x5588c3507dec]
./miniruby(+0x1d0f44)[0x5588c36b2f44]
./miniruby(+0x1e06d9)[0x5588c36c26d9]
./miniruby(+0x1e0c83)[0x5588c36c2c83]
./miniruby(+0x1d8950)[0x5588c36ba950]
./miniruby(+0x1de8cc)[0x5588c36c08cc]
./miniruby(+0x7e529)[0x5588c3560529]
./miniruby(+0x80ba8)[0x5588c3562ba8]
./miniruby(rb_require_safe+0x9)[0x5588c3562ce9]
./miniruby(+0x1d0f44)[0x5588c36b2f44]
./miniruby(+0x1e06d9)[0x5588c36c26d9]
./miniruby(+0x1e0c83)[0x5588c36c2c83]
./miniruby(+0x1d886d)[0x5588c36ba86d]
./miniruby(+0x1de8cc)[0x5588c36c08cc]
./miniruby(+0x7e529)[0x5588c3560529]
./miniruby(+0x7eb96)[0x5588c3560b96]
./miniruby(+0x1d0f44)[0x5588c36b2f44]
./miniruby(+0x1e06d9)[0x5588c36c26d9]
./miniruby(+0x1e0c83)[0x5588c36c2c83]
./miniruby(+0x1d886d)[0x5588c36ba86d]
./miniruby(+0x1de8cc)[0x5588c36c08cc]
./miniruby(+0x78a6a)[0x5588c355aa6a]
./miniruby(ruby_exec_node+0x1d)[0x5588c355cafd]
./miniruby(ruby_run_node+0x1e)[0x5588c355f45e]
./miniruby(+0x2457b)[0x5588c350657b]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1)[0x7f954730f2e1]
./miniruby(_start+0x2a)[0x5588c35065aa]
======= Memory map: ========
5588c34e2000-5588c37ae000 r-xp 00000000 08:05 1848630                    /usr/local/rvm/src/ruby-2.3.0/miniruby
5588c39ad000-5588c39b2000 r--p 002cb000 08:05 1848630                    /usr/local/rvm/src/ruby-2.3.0/miniruby
5588c39b2000-5588c39b3000 rw-p 002d0000 08:05 1848630                    /usr/local/rvm/src/ruby-2.3.0/miniruby
5588c39b3000-5588c39c4000 rw-p 00000000 00:00 0 
5588c5661000-5588c59e7000 rw-p 00000000 00:00 0                          [heap]
7f9540000000-7f9540021000 rw-p 00000000 00:00 0 
7f9540021000-7f9544000000 ---p 00000000 00:00 0 
7f95470d8000-7f95470ee000 r-xp 00000000 08:05 3146773                    /lib/x86_64-linux-gnu/libgcc_s.so.1
7f95470ee000-7f95472ed000 ---p 00016000 08:05 3146773                    /lib/x86_64-linux-gnu/libgcc_s.so.1
7f95472ed000-7f95472ee000 r--p 00015000 08:05 3146773                    /lib/x86_64-linux-gnu/libgcc_s.so.1
7f95472ee000-7f95472ef000 rw-p 00016000 08:05 3146773                    /lib/x86_64-linux-gnu/libgcc_s.so.1
7f95472ef000-7f9547482000 r-xp 00000000 08:05 3150834                    /lib/x86_64-linux-gnu/libc-2.24.so
7f9547482000-7f9547682000 ---p 00193000 08:05 3150834                    /lib/x86_64-linux-gnu/libc-2.24.so
7f9547682000-7f9547686000 r--p 00193000 08:05 3150834                    /lib/x86_64-linux-gnu/libc-2.24.so
7f9547686000-7f9547688000 rw-p 00197000 08:05 3150834                    /lib/x86_64-linux-gnu/libc-2.24.so
7f9547688000-7f954768c000 rw-p 00000000 00:00 0 
7f954768c000-7f954778f000 r-xp 00000000 08:05 3150838                    /lib/x86_64-linux-gnu/libm-2.24.so
7f954778f000-7f954798e000 ---p 00103000 08:05 3150838                    /lib/x86_64-linux-gnu/libm-2.24.so
7f954798e000-7f954798f000 r--p 00102000 08:05 3150838                    /lib/x86_64-linux-gnu/libm-2.24.so
7f954798f000-7f9547990000 rw-p 00103000 08:05 3150838                    /lib/x86_64-linux-gnu/libm-2.24.so
7f9547990000-7f9547998000 r-xp 00000000 08:05 3150836                    /lib/x86_64-linux-gnu/libcrypt-2.24.so
7f9547998000-7f9547b98000 ---p 00008000 08:05 3150836                    /lib/x86_64-linux-gnu/libcrypt-2.24.so
7f9547b98000-7f9547b99000 r--p 00008000 08:05 3150836                    /lib/x86_64-linux-gnu/libcrypt-2.24.so
7f9547b99000-7f9547b9a000 rw-p 00009000 08:05 3150836                    /lib/x86_64-linux-gnu/libcrypt-2.24.so
7f9547b9a000-7f9547bc8000 rw-p 00000000 00:00 0 
7f9547bc8000-7f9547bca000 r-xp 00000000 08:05 3150837                    /lib/x86_64-linux-gnu/libdl-2.24.so
7f9547bca000-7f9547dca000 ---p 00002000 08:05 3150837                    /lib/x86_64-linux-gnu/libdl-2.24.so
7f9547dca000-7f9547dcb000 r--p 00002000 08:05 3150837                    /lib/x86_64-linux-gnu/libdl-2.24.so
7f9547dcb000-7f9547dcc000 rw-p 00003000 08:05 3150837                    /lib/x86_64-linux-gnu/libdl-2.24.so
7f9547dcc000-7f9547e4d000 r-xp 00000000 08:05 275931                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f9547e4d000-7f954804d000 ---p 00081000 08:05 275931                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f954804d000-7f954804e000 r--p 00081000 08:05 275931                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f954804e000-7f954804f000 rw-p 00082000 08:05 275931                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.2
7f954804f000-7f9548067000 r-xp 00000000 08:05 3150849                    /lib/x86_64-linux-gnu/libpthread-2.24.so
7f9548067000-7f9548266000 ---p 00018000 08:05 3150849                    /lib/x86_64-linux-gnu/libpthread-2.24.so
7f9548266000-7f9548267000 r--p 00017000 08:05 3150849                    /lib/x86_64-linux-gnu/libpthread-2.24.so
7f9548267000-7f9548268000 rw-p 00018000 08:05 3150849                    /lib/x86_64-linux-gnu/libpthread-2.24.so
7f9548268000-7f954826c000 rw-p 00000000 00:00 0 
7f954826c000-7f954828f000 r-xp 00000000 08:05 3146707                    /lib/x86_64-linux-gnu/ld-2.24.so
7f9548314000-7f9548415000 rw-p 00000000 00:00 0 
7f9548415000-7f9548466000 r--p 00000000 08:05 787420                     /usr/lib/locale/aa_DJ.utf8/LC_CTYPE
7f9548466000-7f954846a000 rw-p 00000000 00:00 0 
7f9548480000-7f9548481000 rw-p 00000000 00:00 0 
7f9548481000-7f9548482000 ---p 00000000 00:00 0 
7f9548482000-7f9548485000 rw-p 00000000 00:00 0 
7f9548485000-7f954848c000 r--s 00000000 08:05 400247                     /usr/lib/x86_64-linux-gnu/gconv/gconv-modules.cache
7f954848c000-7f954848f000 rw-p 00000000 00:00 0 
7f954848f000-7f9548490000 r--p 00023000 08:05 3146707                    /lib/x86_64-linux-gnu/ld-2.24.so
7f9548490000-7f9548491000 rw-p 00024000 08:05 3146707                    /lib/x86_64-linux-gnu/ld-2.24.so
7f9548491000-7f9548492000 rw-p 00000000 00:00 0 
7ffc46e34000-7ffc47633000 rw-p 00000000 00:00 0                          [stack]
7ffc476a0000-7ffc476a3000 r--p 00000000 00:00 0                          [vvar]
7ffc476a3000-7ffc476a5000 r-xp 00000000 00:00 0                          [vdso]
.
Error running '__rvm_make install',
please read /usr/local/rvm/log/1504112943_ruby-2.3.0/install.log
There has been an error while running make install. Halting the installation.


Can someone help me with this? @pkuczynski

**Comment by mpapis** [comment]
@Corleanus It might be separate problem, please open a new ticket for it.

**Comment by cxsper** [comment]
@nixon-by your solution should be modified, since `gcc5` has been move to AUR repositiories.https://www.archlinux.org/packages/community/x86_64/gcc54/
